libeos-payg/service.c

```diff
@@ -254,25 +262,77 @@ manager_new_cb (GObject      *source_object,
   epg_service_startup_complete (self, task, g_steal_pointer (&provider));
 }
 
+static gboolean
+clock_jump_cb (gpointer user_data)
+{
+  EpgService *self = EPG_SERVICE (user_data);
+  struct timespec ts;
+  gint64 clock_realtime_secs_v1;
+  gint64 clock_boottime_secs_v1;
+  gint64 clock_jump_delta;
+
+  if (G_UNLIKELY (clock_gettime (CLOCK_REALTIME, &ts) != 0))
+    g_error ("clock_gettime() failed for CLOCK_REALTIME: %s", g_strerror (errno));
```

Comment:
This code is copy-pasted 4 times in this file. Perhaps consider a wrapper function which calls `clock_gettime`, `g_error` if it fails, and otherwise returns `tv_sec`?

Reply:
You're right, that's much cleaner